dev/release.sh

```diff
@@ -0,0 +1,614 @@
+It will refuses to work unless the current branch is C<master> or a release
+branch (see L</RELEASE BRANCHES AND TAGS> below for a discussion on those).
+
+B<release.sh> tries to be smart and figure out the next release if no hints
+are given through options, and will exit with an error in ambiguous cases.
+
+B<release.sh> always clones the current workspace into a sub-directory
+named C<< openssl-I<MAJOR>.I<MINOR>.I<PATCH>-tmp >>, where C<< I<MAJOR> >>,
+C<< I<MINOR> >> and C<< I<PATCH> >> are taken from the available version
+information in the source.
+
+=head1 OPTIONS
+
+=over 4
+
+=item B<--major>, B<--minor>
+
+Increment the MAJOR or MINOR version number, and reset the lower numbers
+to zero (for B<--major>, the lower numbers are the I<MINOR> and I<PATCH>
+version numbers, for B<--minor>, the lower numbers is the I<PATCH> version
+number), and state of this branch to become "in development". No actual
+release is done.
+
+=item B<--alpha>, B<--beta>
+
+Set the state of this branch to indicate that alpha or beta releases are
+to be done.
+
+B<--alpha> is only acceptable if the I<PATCH> version number is zero and
+the current state is "in development" or that alpha releases are ongoing.
+
+B<--beta> is only acceptable if the I<PATCH> version number is zero and
+that alpha or beta releases are ongoing.
+
+=item B<--next-beta>
+
+Use together with B<--alpha> to switch to beta releases after the current
+release is done.
+
+=item B<--final>
+
+Set the state of this branch to indicate that regular releases are to be
+done.  This is only valid if alpha or beta releases are currently ongoing.
+
+This implies B<--branch>.
+
+=item B<--branch>
+
+Create a branch specific for the I<MAJOR>.I<MINOR>.x release series, if it
+doesn't already exist, and switch to it.  The exact branch name will be
+C<< openssl-I<MAJOR>.I<MINOR>.x >>.
+
+=item B<--no-upload>
+
+Don't upload the produce files.
+
+=item B<--no-update>
+
+Don't run C<make update>.
+
+=item B<--verbose>
+
+Verbose output.
+
+=item B<--debug>
+
+Display extra debug output.  Implies B<--no-upload>
+
+=item B<--force>
+
+Force execution.  Precisely, the check that the current branch is C<master>
+or a release branch is not done.
+
+=item B<--help>
+
+Display a quick help text and exit.
+
+=item B<--manual>
+
+Display this manual and exit.
+
+=back
+
+=head1 RELEASE BRANCHES AND TAGS
+
+Prior to OpenSSL 3.0, the release branches were named
+C<< OpenSSL_I<version>-stable >>, and the release tags were named
+C<< OpenSSL_I<version> >> for regular releases, or
+C<< OpenSSL_I<version>-preI<n> >> for pre-releases.
+
+From OpenSSL 3.0 ongoing, the release branches are named
+C<< openssl-I<MAJOR>.I<MINOR>.x >>, and the release tags are named
+C<< openssl-I<MAJOR>.I<MINOR>.I<PATCH> >> for regular releases, or
+C<< openssl-I<MAJOR>.I<MINOR>.I<PATCH>-alphaI<n> >> for alpha releases
+and C<< openssl-I<MAJOR>.I<MINOR>.<PATCH>-betaI<n> >> for beta releases.
```

Comment:
Why the change of" OpenSSL_" to "openssl-"?

Reply:
Like I said in https://github.com/openssl/openssl/pull/11516/files#r406868833, having tags match the tar file could be beneficial.